coqlspclient/proof_file.py

```diff
@@ -158,53 +161,67 @@ def get_context(file_path: str, timeout: int):
                 v_file = aux_file.get_diagnostics(
                     "Locate Library", library, last_line + i + 1
                 ).split()[-1][:-1]
-                coq_file = CoqFile(v_file, library=library, timeout=timeout)
-                coq_file.run()
+
+                with open(v_file, "r") as f:
+                    hash = hashlib.md5(f.read().encode("utf-8")).hexdigest()
+                if (v_file, hash) in _AuxFile.__CACHE:
+                    aux_context = _AuxFile.__CACHE[(v_file, hash)]
+                else:
+                    coq_file = CoqFile(v_file, library=library, timeout=timeout)
+                    coq_file.run()
+                    aux_context = coq_file.context
+                    _AuxFile.__CACHE[(v_file, hash)] = aux_context
+                    coq_file.close()
 
                 # FIXME: we ignore the usage of Local from imported files to
                 # simplify the implementation. However, they can be used:
                 # https://coq.inria.fr/refman/language/core/modules.html?highlight=local#coq:attr.local
-                for term in list(coq_file.context.terms.keys()):
-                    if coq_file.context.terms[term].text.startswith("Local"):
-                        coq_file.context.terms.pop(term)
-
-                context.update(**vars(coq_file.context))
-                coq_file.close()
+                for term in list(aux_context.terms.keys()):
+                    if aux_context.terms[term].text.startswith("Local"):
+                        aux_context.terms.pop(term)
+                context.update(**vars(aux_context))
 
         return context
 
 
-class ProofState(object):
+class ProofFile(CoqFile):
     """Allows to get information about the proofs of a Coq file.
     ProofState will run the file from its current state, i.e., if the file
     has finished its execution, ProofState won't return anything. The Coq file
     will be fully checked after the creation of a ProofState.
 
+    FIXME
```

Comment:
Fix what? :eyes:

Reply:
I forgot to update the attributes in the comment